builtin/v13/migration/market.go

```diff
@@ -318,8 +318,10 @@ func (m *marketMigrator) migrateProviderSectorsAndStatesFromScratch(ctx context.
 		// Refresh proposals array periodically to avoid holding onto all ~10GB of memory
 		// throughout whole migration.
 		// This has limited impact on caching speedups because the access pattern is sequential.
+		// The minimum batch-size is 100 (hence a 100,000 minimum proposals before this
+		// kicks in).
 		cnt += 1
-		if proposalsSize > 0 && cnt%(int(proposalsSize/1000)) == 0 {
+		if proposalsSize > 100_000 && cnt%(int(proposalsSize/1000)) == 0 {
```

Comment:
note: I'm not sure this is how we want to write this anyways. IMO, we want:

```suggestion
		if cnt%K == 0 {
```

where `K` is an experimentally derived number that keeps the memory below some set limit.

(not worth changing now, honestly)